tests/karate/src/test/java/be/utils/JsonConverter.java

```
@@ -0,0 +1,160 @@
+        privateKeyBytes[i] = (byte) val;
+      }
+      PublicKeySign publicKeySign = new Ed25519Sign(privateKeyBytes);
+      byte[] singie = publicKeySign.sign(messageData.getBytes(StandardCharsets.UTF_8));
+      return Base64.getUrlEncoder().encodeToString(singie);
+
+    } catch (Exception e) {
+      System.out.println("CANNOT SIGN");
+      e.printStackTrace();
+    }
+    // This should never happen
+    return signatureForced;
+  }
+
+  /**
+   * If want to test a sender that is not the organizer we can change the sender public key
+   */
+  public void setSenderPk(String newSenderPk) {
+    this.senderPk = newSenderPk;
+  }
+
+  /**
+   * If we want to test having a secret key that does not match the private key we can set a different private key
+   */
+  public void setSenderSk(String newSenderSkHex) {
+    this.privateKeyHex = newSenderSkHex;
+  }
+
+  /**
+   * If we want to test having a signature that does not match the data and private key we can set it by force
+   */
+  public void setSignature(String newSignature) {
+    this.signatureForced = newSignature;
+  }
+
+  /**
+   * If we want to test how the backend behaves with a non-valid message id we can set it by force
+   */
+  public void setMessageIdForced(String messageIdForced) {
+    this.messageIdForced = messageIdForced;
+  }
+
+  /**
+   * Hashes an arbitrary number of arguments
+   */
+  public String hash(byte[]... allData) {
+    try {
+      MessageDigest digest = MessageDigest.getInstance("SHA-256");
+      for (byte[] data : allData) {
+        String dataLength = Integer.toString(data.length);
+        digest.update(dataLength.getBytes());
+        digest.update(data);
+      }
+      return Base64.getUrlEncoder().encodeToString(digest.digest());
+    } catch (Exception e) {
+      e.printStackTrace();
+      System.out.println("Exception happened in hash construction");
+    }
+    return "Hash is not constructed correctly";
```

I would prefer to throw an error rather than returning an invalid value.

You can even throw the error and remove the try block.